users/views.py

```diff
@@ -79,29 +81,91 @@ def post(self, request):
-    else:
-        email = user.email
-        otp = send_email_otp([email])
-        user.otp = otp
-        user.save()
-        message = "An otp has been sent to your mobile no to reset your password"
-        res_status = status.HTTP_200_OK
+class ForgetPasswordView(APIView):
+
+    serializer_class = ForgetPasswordSerializer
+    
+    @swagger_auto_schema(
+        operation_id='forget_password',
+        request_body=ForgetPasswordSerializer,
+        responses={
+            '200': set_example(responses.forget_password_200),
+            '400': set_example(responses.forget_password_400),
+            '404': set_example(responses.login_404),
+        },
+    )
+    def post(self, request):
+        """
+        Forgot Password API where the email is posted and OTP is sent to the user.
+        """
+
+        serializer = ForgetPasswordSerializer(data=request.data)
+
+        # Checking if the email entered is valid.
+        if serializer.is_valid():
+            valid_data = serializer.validated_data
+
+            # Check if such a user exists.
+            try:
+                user = CustomUser.objects.get(email=valid_data['email'])
+            except CustomUser.DoesNotExist:
+                return Response("Account with this email id doesn't exists. Kindly signup.", status.HTTP_404_NOT_FOUND)
+            else:
+                otp = str(randint(1000, 9999))
+                user.otp = otp
+                user.save()
+                send_email_otp(recipient_list=[user.email], otp=otp)
+
+                return Response("An otp has been sent to you to reset your password", status.HTTP_200_OK)
+
+        # If the email entered was invalid or empty.
+        else:
+            data = serializer.errors
+            return Response(data, status.HTTP_400_BAD_REQUEST)
+
+class ChangePasswordView(APIView):
+    
+    serializer_class = ChangePasswordSerializer
+
+    @swagger_auto_schema(
+        operation_id='change_password',
+        request_body=ChangePasswordSerializer,
+        responses={
+            '200': set_example(responses.change_password_200),
+            '400': set_example(responses.change_password_400),
+            '404': set_example(responses.change_password_404),
+            '401': set_example(responses.change_password_401),
+        },
+    )
+
+    def post(self, request):
+        '''
+        Change password API is where the email, otp and password is posted and password is changed
+        '''
+
+        serializer = ChangePasswordSerializer(data=request.data)
+
+        if serializer.is_valid():
+            valid_data = serializer.validated_data
+
+            # Check if such a user exists.
+            try:
+                user = CustomUser.objects.get(email=valid_data['email'])
+            except CustomUser.DoesNotExist:
+                return Response("Account with this email id doesn't exists. Kindly signup.", status.HTTP_404_NOT_FOUND)
```

Comment:
```suggestion
                return Response(responses.change_password_404, status.HTTP_404_NOT_FOUND)
```
I think responses should be used, I don't know if we just return the message directly.